content/technotes/2020-07-09-cran-checks-update/index.Rmd

```diff
@@ -0,0 +1,138 @@
+- platforms: platform the status occurs on, including negation (e.g., “-solaris”). options: solaris, osx, linux, windows, devel, release, patched, oldrel
+- regex: a regex to match against the text of an error in `check_details.output`
+
+A user can set as many rules as they like. A package can have more than one rule. Users can delete only their own rules (as one would expect). Rules are used indefinitely, until deleted by the user. 
+
+Once a rule is triggered an email is sent to the user. We won't send the same email for 5 days. After 5 days has past, if the rule is matched, we'll send the same email; and repeat. 
+
+Feel free to ignore these emails, or act them as you see fit. 
+
+The email is structured as follows:
+
+{{<figure src="email.png" alt="example cran checks email" width="450">}}
+
+- Triggered rule: with a list of each of the four categories and their values.
+- Date of the check result (matches closely the date on the CRAN check results that CRAN maintainers created)
+- Your check results: link to the CRAN checks API JSON output for your package
+- The rest is information, report bugs, ask for help, etc. Note the "Unsubscribe" link doesn't actually work yet!
+
+#### Managing notifications
+
+There is no web interface to notifications at this time. The only interface is the cchecks R package.
+
+First, you'll need to register for a token (aka key). Using [cchn_register()](https://docs.ropensci.org/cchecks/reference/cchn_register), you run the function with or without an email address. If no email address is given we look for an email address in various places.
+
+```r
+cchn_register()
+```
+
+Running `cchn_register()` caches the token in a file locally on your computer. 
+
+After registering, you can manage rules for your packges. There's two ways to manage rules: a) across packages, or b) in a single package context.
+
+Functions prefixed with `cchn_pkg_` operate within a package directory. That is, your current working directory is an R package, and is the package for which you want to handle CRAN checks notifications. These functions make sure that you are inside of an R package, and use the email address and package name based on the directory you're in.
+
+Functions prefixed with just `cchn_` do not operate within a package. These functions do not guess package name at all, but require the user to supply a package name (for those functions that require a package name); and instead of guessing an email address from your package, we guess email from the cached email/token file.
+
+If you don't use the package specific functions, you can add a rule like:
+
+```r
+cchn_rule_add(package = "foobar", status = "warn", platform = 2)
+```
+
+Using package specific functions, you can add a rule like:
+
+```r
+cchn_pkg_rule_add(status = "warn", platform = 2)
+```
+
+See `cchn_rules_add()` for adding many rules at once.
+
+<br>
+
```

Comment:
What I'd also like to read about in this section is some tips for how to define rules: what would you recommend? Could you give your own rules as an example?